src/app/reporting/Engine.cpp

```diff
@@ -512,11 +513,18 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
 
     VerifyOrExit(apReadHandler != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(apReadHandler->GetSession() != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    if (apReadHandler->ShouldUseLargePayloadBuffer())
```

Comment:
How about:

```cpp
const size_t maxSduSize = apReadHandler->ShouldUseLargePayloadBuffer()
    ? chip::app::kMaxLargeSecureSduLengthBytes
    : chip::app::kMaxSecureSduLengthBytes;
```

To make the 2 constants clearer by having them close together (and also making the value constant)? also see note above: could we make ReadHandler have this logic instead of passing down a bool around?

Reply:
ReadHandler now provides a function to infer this internally from the member session handle.